tools/integration-tests/tester/tests/testutil.go

```diff
@@ -619,3 +619,52 @@ func txMetadataStateEqual(t *testing.T, node *framework.Node, txID string, expIn
 	}
 	return true, metadata.GradeOfFinality
 }
+
+// TryConfirmMessage tries to confirm the message on all the peers provided within the time limit provided.
+func TryConfirmMessage(t *testing.T, n *framework.Network, requiredPeers []*framework.Node, msgID string, waitFor time.Duration, tick time.Duration) {
+	var peers []*framework.Node
+	for _, peer := range n.Peers() {
+		if _, err := peer.GetMessage(msgID); err == nil {
+			peers = append(peers, peer)
+		}
+	}
+	if len(peers) == 0 {
+		log.Println("msg ", msgID, "is not found on any node")
+		t.FailNow()
+	}
+
+	confirmedOnAllPeers := func(msgID string, peers []*framework.Node) bool {
```

Comment:
I think this method can be extracted as a general function to check if a certain message is confirmed on all supplied peers.